modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/TypeWrapper.java

```diff
@@ -76,7 +76,7 @@ public TypeWrapper(ResolvableType resolvableType, ClassTypeLoader typeLoader) {
       resolvableTypeGenerics = resolvableType.getGenerics();
       for (ResolvableType type : resolvableTypeGenerics) {
         TypeWrapper concreteType;
-        if (type == resolvableType) {
+        if (type == resolvableType || (type.isAssignableFrom(resolvableType) && resolvableType.isAssignableFrom(type))) {
```

Comment:
are enums the only use case for this? if so, why not replacng the second condition with one that checks for an enum?

Reply:
it affects more uses than just enums, added another test case to show this